homeassistant/components/zwave/node_entity.py

```diff
@@ -95,6 +96,14 @@ def __init__(self, node, network):
         dispatcher.connect(
             self.network_scene_activated, ZWaveNetwork.SIGNAL_SCENE_EVENT)
 
+    @property
+    def unique_id(self):
+        """Unique ID of Z-wave node."""
+        return '{}-{}-{}'.format(
+            self.node_id,
+            slugify(self._manufacturer_name),
```

Comment:
Don't manufacturer name and product name need to be fetched and thus it might be that a device gets a new unique ID after restart? Also, why wouldn't node id be enough ? And what about the devices like the 5-in-1 sensor, do they share a node id ?

Reply:
There are 2 options:
1) Use node_id only: The unique_id will indeed be stable, but the default entity_id will be "_" (as both manufacturer and product are empty). So this bad entity_id will be kept even after restart.
2) Use node_id, manufacturer and product. unique_id changes after restart getting a proper entity_id, but leaving a pollution in entity registry.

I think the best solution (out of scope of this PR) would be for zwave support for removing + re-adding the entity upon a change in manufacturer/product. This will allow properly adding zwave devices without restart.

Reply:
We should just depend on node id. 

I don't like the pollution of the entity registry because what if a user starts configuring the devices? Renaming it, changing entity id (Future pr). It restarts and everything is lost.

How long does it take for Z-Wave to be aware of the manufacturer and product name? Wouldn't it make more sense to not add the entity to Home Assistant until manufacturer and product name are loaded? We could show a notification in the meanwhile to make sure that we notify the user that a device is pending.

Reply:
I should be quite fast (a few seconds) in the regular case of adding 1 powered device to an otherwise "ready" network.
There is also a case of losing zw_cfg file and restarting without it - in that case it would be a while: minutes, depending on network size, hours if waiting for unpowered devices to wake up.

The problem is I have trouble testing this (wait for data fetch) as my network zwave is quite large and each restart takes ages.

Maybe zwave should only report unique_id if manufacturer and product are non-empty? This will prevent pollution and keep things as-is (without registry) until data is properly fetched.

Reply:
Can we do a mix of both? We wait up to 30 seconds to see if manufacturer/product name arrive before adding it to Home Assistant. After 30 seconds we add it but it won't have a unique ID.